scripts/docs/to-integrations-repo_v2

```diff
@@ -0,0 +1,312 @@
+            processed_config.update(process_config_fields_helper(field.get("elementStruct"), field.get("yamlName")))
+
+    # Return empty dictionary if there are no config options
+    if len(processed_config.get(yamlname)) == 0:
+        processed_config.clear()
+
+    return processed_config
+
+
+def process_config_fields(config):
+    return process_config_fields_helper(config)
+
+
+def monitor_docs_per_integrations_repo(monitor_docs):
+    """s
+     Returns a dict which is a processed version of
+     selfdescribe for easy consumption by Jinja2 template
+    """
+
+    # {% set required = "no" %}
+    # # {% if field.required %}
+    # # {% set required = "**yes**" %}
+    # # {% endif %}
+    # {% set default = "" %}
+    # {% if field.default %}
+    # {% set default = "(**default:**) `{}`.format(field.default) %}
+    # {% endif %}
+
+    out = {}
+
+    for monitor_doc in monitor_docs:
+        integrations_dir = MONITORS_TO_INTEGRATIONS_DIR.get(
+            (monitor_doc["config"]["package"], monitor_doc["monitorType"]), None)
+
+        if not integrations_dir:
+            continue
+
+        if integrations_dir not in out:
+            out[integrations_dir] = {}
+            out[integrations_dir]["monitor_types"] = []
+            out[integrations_dir]["send_all_metrics"] = OrderedDict()
+            out[integrations_dir]["description"] = OrderedDict()
+            out[integrations_dir]["configuration"] = OrderedDict()
+            out[integrations_dir]["metrics"] = OrderedDict()
+            out[integrations_dir]["dimensions"] = OrderedDict()
+            out[integrations_dir]["properties"] = OrderedDict()
+
+        monitor_type = monitor_doc["monitorType"]
+        out[integrations_dir]["monitor_types"].append(monitor_type)
+        out[integrations_dir]["send_all_metrics"].update({monitor_type: monitor_doc["sendAll"]})
+        out[integrations_dir]["description"].update({monitor_type: monitor_doc["doc"]})
+        out[integrations_dir]["configuration"].update({monitor_type: process_config_fields(monitor_doc["config"])})
+        out[integrations_dir]["metrics"].update(
+            process_metrics_from_self_describe(monitor_type, monitor_doc["metrics"]))
+        out[integrations_dir]["dimensions"].update({monitor_type: monitor_doc["dimensions"]})
+        out[integrations_dir]["properties"].update({monitor_type: monitor_doc["properties"]})
+
+    return out
+
+
+def get_jinja_env():
+    j2_env = Environment(loader=FileSystemLoader(str(INTEGRATIONS_DOC_TEMPLATE_DIR)), trim_blocks=True)
+
+    # Add custom filters
+    j2_env.filters["surround_by_tick"] = surround_by_tick
+    j2_env.filters["has_atleast_one_monitor_with_section"] = has_atleast_one_monitor_with_section
+    j2_env.filters["send_all_metrics_from_all_monitors"] = send_all_metrics_from_all_monitors
+
+    return j2_env
+
+
+def get_template():
+    j2_env = get_jinja_env()
+    template = j2_env.get_template("integrations_doc.md.jinja2")
```

Comment:
Make sure this script will work the same no matter what the PWD is when run.

Reply:
A file system loader object is passed to the environment `j2_env`. Any template in the path passed into the filesystem loader object, can be accessed in this manner. `INTEGRATIONS_DOC_TEMPLATE_DIR` is passed in and is derived from the current script directory. That should suffice right?

Reply:
I suppose so, just make sure that you can run the script equally well when your shell is in the root of the repo as from the same directory where the script is.